docs/source/about_mapstyle_vs_iterable.mdx

```diff
@@ -0,0 +1,220 @@
+```python
+data_files = {"train": ["path/to/data.csv"]}
+my_dataset = load_dataset("csv", data_files=data_files, split="train")
+print(my_dataset[0])
+```
+
+However here it requires a conversion step from CSV to Arrow format, which can take time and disk space if your dataset is big.
+
+To save disk space ans skip the conversion step, you can define an iterable dataset by streaming from the local files directly.
+This way the data is read progessively from the local files as you iterate over the dataset:
+
+```python
+data_files = {"train": ["path/to/data.csv"]}
+my_iterable_dataset = load_dataset("csv", data_files=data_files, split="train", streaming=True)
+for example in my_iterable_dataset:  # this reads the CSV file progressively as you iterate over the dataset
+    print(example)
+    break
+```
+
+Many files formats are supported like CSV, JSONL, Parquet, images, audio files etc.
+You can find more information in the corresponding guides for loading [tabular](./tabular_load), [text](./nlp_load), [vision]()./image_load) and [audio](./audio_load]) datasets.
+
+## Eager data processing and lazy data processing
+
+When you process a [`Dataset`] object using [`Dataset.map`] for example, the data processing happens immediatly when you call [`Dataset.map`] and returns the processed dataset.
+This is similar to how `pandas` works for example.
+
+```python
+my_dataset = my_dataset.map(process_fn)  # process_fn is applied on all the examples of the dataset
+print(my_dataset[0])
+```
+
+On the other hand, due to the "lazy" nature of iterable datasets, calling [`IterableDataset.map`] does not iterate over the full dataset to apply your `map` function.
+Instead, your `map` function is applied on-the-fly when you iterate over the iterable dataset.
+
+Because of that, you can chain multiple processing steps and they will all run at once when you start iterating over the dataset:
+
+```python
+my_iterable_dataset = my_iterable_dataset.map(process_fn_1)
+my_iterable_dataset = my_iterable_dataset.filter(filter_fn)
+my_iterable_dataset = my_iterable_dataset.map(process_fn_2)
+
+# process_fn_1, filter_fn and process_fn_2 are applied on-the-fly when iterating over the dataset
+for example in my_iterable_dataset:  
+    print(example)
+    break
+```
+
+## Exact and fast approximate shuffling
+
+When you shuffle a [`Dataset`] using [`Dataset.shuffle`], you apply an exact shuffling of the dataset.
+It works by taking a list of indices `[0, 1, 2, ... len(my_dataset) - 1]` and shuffling this list.
+Then, accessing `my_dataset[0]` accesses the row and the index defined by the first element of the indices mapping that has been shuffled:
+```python
+my_dataset = my_dataset.shuffle(seed=42)
+print(my_dataset[0])
+```
+
+Since we don't have random access to the rows in the case of iterable datasets, we can't use a shuffled list of indices and access a row at an arbitrary position.
```

Comment:
```suggestion
Since we don't have random access to the rows in the case of an `IterableDataset`, we can't use a shuffled list of indices and access a row at an arbitrary position.
```